chainerx_cc/chainerx/routines/connection.cc

```diff
@@ -393,6 +394,32 @@ Array Linear(const Array& x, const Array& w, const absl::optional<Array>& b, uin
     return out_matrix.Reshape(out_shape);
 }
 
+Array EmbedId(const Array& x, const Array& w, absl::optional<int64_t> ignore_label) {
+    if (ignore_label.has_value() && (*ignore_label < 0 || *ignore_label >= w.shape()[0])) {
+        throw IndexError{"ignore_label should be greater than equal to 0 and less than" + w.shape()[0]};
+    }
+    StackVector<int64_t, kMaxNdim> shape_vec;
+    shape_vec.emplace_back(w.shape()[1]);
+    Shape shape{shape_vec};
+    std::vector<Array> ignore_list;
+    std::vector<Array> x_split = Split(x, x.shape()[0], 0);
```

Comment:
These variables are not used in `ignore_label == nullopt` case.
Let's move it within the `if` clause.